docs/arcadia-users-group/20221017-conda/lesson.md

```diff
@@ -0,0 +1,369 @@
+
+
+<center>
+<figure markdown>
+  ![condachannels2](conda_channels2.png){ width="700" }
+  <figcaption> Cartoon of conda channel priorities by Gergely Szerovay <a href='https://www.freecodecamp.org/news/why-you-need-python-environments-and-how-to-manage-them-with-conda-85f155f4353c/' target='_blank'>www.freecodecamp.org</a> </figcaption>
+</figure>
+</center>
+
+
+### Creating a new environment
+
+Other than mamba, we recommend you keep you `base` environment free from all other installations. 
+If you install everything in `base`, you'll end up with lots of dependencies that need to be solved which will decrease installation speeds.
+It also becomes more difficult to document software installations and versions required for specific project. 
+
+To create a new environment named, for instance `mynewenv` (you can name it what ever you like), run:
+
+```
+conda create --name mynewenv
+```
+
+### Navigating environments through `activate` and `deactivate`
+
+If you want to use an environment (for instance manage packages, or run Python scripts inside it) you need to first activate it.
+
+```
+conda activate mynewenv
+```
+
+The command prompt should now be prepended by `(mynewenv)` instead of `(base)`.
+The command prompt is a useful tool to orient yourself as to which environment you're currently using.
+
+If you install a new tool while in an environment, it only gets installed in that environment.
+```
+mamba install samtools
+```
+
+Even better, you can specify which version of samtools you would like to install:
+```
+mamba install samtools=1.9
+```
+
+We just installed a very old version. 
+We can ask mamba to automatically update our package installation to the latest version:
+```
+mamba update samtools
+```
+
+The directories of the active environment’s executable files are added to the system path (this means that you can now access them more easily). 
+
+```
+which samtools
+```
+
+You can leave an environment with this command:
+
+```
+conda deactivate
+```
+
+Now, if we run the same command, we'll get a different result:
+
+```
+which samtools
+```
+
+### Searching, installing, and removing packages
+
+To list out all the installed packages in the currently active environment, run:
+
+```
+conda list
+```
+
+To search for all the available versions of a certain package, you can use the search command. 
+For instance, to list out all the versions of samtools, run:
+
+```
+conda search samtools
+```
+
+Similarly to the conda list command, this one results in a list of the matching package names, versions, and channels:
+
+```
+Loading channels: done
+# Name                       Version           Build  Channel
+samtools                      0.1.12               0  bioconda
+samtools                      0.1.12               1  bioconda
+samtools                      0.1.12               2  bioconda
+samtools                      0.1.12      hdd8ed8b_3  bioconda
+```
+
+### Freezing an environment
```

Comment:
Beautiful

Reply:
this is great - will be helpful for a lot of people